src/Helpers/Utils.php

```diff
@@ -129,9 +129,20 @@ public static function removeBackslashes($data)
         return $data;
     }
 
+    /**
+     * @unreleased
+     */
+    public static function recursiveUrlDecode($data)
+    {
+        $decoded = urldecode($data);
+
+        return $decoded === $data ? $data : self::recursiveUrlDecode($decoded);
```

Comment:
What's the reason we need to url decode recursively? It would be helpful to add a comment with your reason here.